File: backend/src/main/java/com/pl/shugo/gsolog/application/service/AdminService.java
package com.pl.shugo.gsolog.application.service;

import com.pl.shugo.gsolog.domain.entity.User;
import com.pl.shugo.gsolog.domain.repository.UserRepository;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;

/**
 * Admin service for user management operations.
 * Only accessible by ADMIN role.
 */
@Service
public class AdminService {

    private final UserRepository userRepository;

    public AdminService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    /**
     * Get all users in the system.
     * No filtering by userId - admins can see all users.
     *
     * @return Flux of all users
     */
    public Flux<User> getAllUsers() {
        return userRepository.findAll();
    }
}

File: frontend/src/app/core/services/admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { UserResponse } from '../models/auth.models';

/**
 * Admin service for user management operations.
 */
@Injectable({
  providedIn: 'root'
})
export class AdminService {
  private apiUrl = environment.apiBaseUrl + '/admin';

  constructor(private http: HttpClient) {}

  /**
   * Get all users (ADMIN only).
   */
  getAllUsers(): Observable<UserResponse[]> {
    return this.http.get<UserResponse[]>(`${this.apiUrl}/users`);
  }
}
